src/PaginatedData.elm

```diff
@@ -360,350 +423,181 @@ where the total count is zero. So, if `totalCount` remains a `Nothing`, we would
-    in
-    EveryDict.insert identifier (RemoteData.Success existingDataAndPagerUpdated) dict
+{-| Mark that a request for the specified page (1-based) is currently in progress.
+-}
+setPageAsLoading : Int -> PaginatedData err key value -> PaginatedData err key value
+setPageAsLoading pageNumber =
+    handleFetchedPage pageNumber Loading
 
 
-{-| Used to indicate we're loading a page for the first time.
--}
-setPageAsLoading :
-    identifier
-    -> Int
-    -> EveryDict identifier (WebData (PaginatedData key value))
-    -> EveryDict identifier (WebData (PaginatedData key value))
-setPageAsLoading identifier pageNumber dict =
-    let
-        existing =
-            EveryDict.get identifier dict
-                |> Maybe.withDefault (RemoteData.Success emptyPaginatedData)
+{-| When you receive a response from the backend to your request for a page of
+data, use this function to update the `PaginatedData` with the response.
 
-        existingDataAndPager =
-            existing
-                |> RemoteData.toMaybe
-                |> Maybe.withDefault emptyPaginatedData
+  - The first parameter is the page number which you requested (1-based).
 
-        pagerUpdated =
-            EveryDict.insert pageNumber RemoteData.Loading existingDataAndPager.pager
+  - The second parameter is the response received from the backend.
 
-        existingDataAndPagerUpdated =
-            { existingDataAndPager | pager = pagerUpdated }
-    in
-    EveryDict.insert identifier (RemoteData.Success existingDataAndPagerUpdated) dict
+    In the `Success` case, the `EveryDictList` is the data you received, and
+    the `Int` is the total count of all items available on the backend (not
+    just the items on this page). We will update the `PaginatedData` to reflect
+    the successful request.
+
+    If you provide a `Failure` response, we will update the page to record the
+    failed request.
 
+    If you provide `Loading` or `NotAsked`, we will return the `PaginatedData`
+    unchanged. (You can use `setPageAsLoading` to return a page to `Loading`
+    status.)
 
-{-| Insert multiple Items into the data and pager dict.
 -}
-insertMultiple :
-    identifier
-    -> Int
-    -> RemoteData.RemoteData e ( EveryDictList k v, Int )
-    -> (number -> a1)
-    -> (( k, v ) -> Maybe a1)
-    -> (k -> v -> EveryDictList a1 value -> EveryDictList a1 value)
-    -> (k -> v -> ( a1, EveryDictList a1 value ) -> ( a1, EveryDictList a1 value ))
-    -> EveryDict identifier (RemoteData.RemoteData e (PaginatedData a1 value))
-    -> EveryDict identifier (RemoteData.RemoteData e (PaginatedData a1 value))
-insertMultiple identifier pageNumber webdata defaultItemFunc getItemFunc insertFunc insertAfterFunc dict =
+handleFetchedPage :
+    Int
+    -> RemoteData err ( EveryDictList key value, Int )
+    -> PaginatedData err key value
+    -> PaginatedData err key value
+handleFetchedPage pageNumber webdata (PaginatedData existing) =
```

Comment:
should we bail out if `pageNumber < 1` here as-well?

Reply:
Yes, that would make sense.